### PR TITLE
Add uncopyable.h, class uncopyable

### PR DESCRIPTION
**Pull request by rolandreichweinbmw**

### CODE REVIEW

**Comment by rolandreichweinbmw** [comment]
I don't see a way to add a unit test for this feature. It's a convenience feature which being applied incorrectly would not compile.

**Comment by rolandreichweinbmw** [comment]
If you prefer a unit test including a dummy use of this class (non-functional feature), I can add it.

File: include/etl/uncopyable.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  /// Can be used to make a class uncopyable by deriving from it.
+  //***************************************************************************
+  class uncopyable
+  {
+  public:
+    uncopyable(uncopyable const&)            = delete;
+    uncopyable& operator=(uncopyable const&) = delete;
+
+  protected:
+    uncopyable() = default;
```
Comment (jwellbelove) [inline include/etl/uncopyable.h:50]: I've see comments online that Scott Meyers also adds the destructor in his 'Effective C++' book.

Comment (rolandreichweinbmw) [reply include/etl/uncopyable.h:50]: I.e. a non-virtual public "=default" destructor?

Is there a reasoning behind it?

Comment (jwellbelove) [reply include/etl/uncopyable.h:50]: It looks like it is intended to make the destructor `protected`, rather than default `public`.

**Comment by jwellbelove** [comment]
I have sometimes added a commented out unit test that contains 'uncompilable' code, with a surrounding comment that the test should fail to compile if uncommented.
It can be useful if you want to check that the disabled functionality is _still_ disabled after a code change.

**Comment by jwellbelove** [comment]
I also sometimes do this for `static_assert` checks.

**Comment by rolandreichweinbmw** [comment]
I just added the destructor as discussed, and unit tests, including disabled code that intentionally wouldn't compile.

File: include/etl/uncopyable.h (hunk 1 above)
Comment (rolandreichweinbmw) [reply include/etl/uncopyable.h:50]: Added the destructor.

**Pull request merged**